python-test-samples/apigw-lambda-dynamodb/tests/integration/test_api_gateway.py

```diff
@@ -0,0 +1,95 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import os
+from unittest import TestCase
+from boto3.dynamodb.conditions import Key
+import boto3
+import requests
+
+"""
+Set the environment variable AWS_SAM_STACK_NAME 
+to match the name of the stack you will test
+
+AWS_SAM_STACK_NAME=<stack-name> python -m pytest -s tests/integration -v
+"""
+
+class TestApiGateway(TestCase):
+    api_endpoint: str
+
+    @classmethod
+    def get_stack_name(cls) -> str:
+        stack_name = os.environ.get("AWS_SAM_STACK_NAME")
+        if not stack_name:
+            raise Exception(
+                "Cannot find env var AWS_SAM_STACK_NAME. \n"
+                "Please setup this environment variable with the stack name where we are running integration tests."
+            )
+
+        return stack_name
+
+    def setUp(self) -> None:
+        """
+        Based on the provided env variable AWS_SAM_STACK_NAME,
+        We use the cloudformation API to retrieve the HelloPersonApi URL and the DynamoDB Table Name
+        We also seed the DynamoDB Table for the test
+        """
+        stack_name = TestApiGateway.get_stack_name()
+
+        client = boto3.client("cloudformation")
+
+        try:
+            response = client.describe_stacks(StackName=stack_name)
+        except Exception as e:
+            raise Exception(
+                f"Cannot find stack {stack_name}. \n" f'Please make sure stack with the name "{stack_name}" exists.'
+            ) from e
+
+        # HelloPersonApi
+        stack_outputs = response["Stacks"][0]["Outputs"]
+        api_outputs = [output for output in stack_outputs if output["OutputKey"] == "HelloPersonApi"]
+        self.assertTrue(api_outputs, f"Cannot find output HelloPersonApi in stack {stack_name}")
+        self.api_endpoint = api_outputs[0]["OutputValue"]
+
+        # DynamoDBTableName
+        dynamodb_outputs = [output for output in stack_outputs if output["OutputKey"] == "DynamoDBTableName"]
+        self.assertTrue(dynamodb_outputs, f"Cannot find output DynamoDBTableName in stack {stack_name}")
+        self.dynamodb_table_name = dynamodb_outputs[0]["OutputValue"] 
+
+        # Seed the DynamoDB Table with Test Data
+        dynamodb_resource = boto3.resource("dynamodb", region_name="us-east-1")
+        dynamodb_table = dynamodb_resource.Table(name=self.dynamodb_table_name)
+        dynamodb_table.put_item(Item={"PK": "TEST001", 
+                                            "SK": "NAME#",
+                                            "data": "Unit Test Name Data"})
+
+
+    def tearDown(self) -> None:
+        """
+        # For tear-down, remove any data injected for the tests
+        # Table particular care to ensure these values are unique and identifiable as TEST data.
```

Comment:
(minor) Table => Take?

Reply:
Thank you for catching the typo!  I've staged a fix for the next commit.